src/compiler/transformers/declarations/types.ts

```diff
@@ -1,51 +1,21 @@
 import {
-    AccessorDeclaration,
-    AllAccessorDeclarations,
-    AnyImportSyntax,
-    ComputedPropertyName,
-    Declaration,
-    ElementAccessExpression,
-    EntityNameOrEntityNameExpression,
-    EnumMember,
-    Expression,
-    FunctionDeclaration,
-    ImportDeclaration,
-    LateBoundDeclaration,
-    Node,
-    ParameterDeclaration,
-    PropertyAccessExpression,
-    PropertyDeclaration,
-    PropertySignature,
-    ResolutionMode,
-    SignatureDeclaration,
-    StringLiteralLike,
-    Symbol,
-    SymbolTracker,
-    SymbolVisibilityResult,
-    VariableDeclaration,
+    CompilerOptions,
+    CoreEmitResolver,
+    CoreTransformationContext,
+    Diagnostic,
+    NodeFactory,
 } from "../../_namespaces/ts";
 
+/** @internal */
+export interface IsolatedTransformationContext extends CoreTransformationContext {

```

Comment:
Why does this need its own version of `TransformationContext` if both are marked `@internal`? `TransformationContext` already has `getEmitResolver()`, `getCompilerOptions()`, `factory`, and `addDiagnostic`. If the only reason is the `useLocalInferenceTypePrint` field, then I would argue that it doesn't belong on the context.

Reply:
Yes but `IsolatedTransformationContext` provides an emit resolver that has fewer features than the full resolver (just the `CoreEmitResolver`). I would ideally want this to be obvious in the declaration transform. 

We could add implementations that just throw for everything that is not supported, but that would be hard to follow and prone to mistakes in the declaration transform (someone might accidentally call one of these un-implemented methods)

Right now we have a discriminated union in declaration emit that ensures you need to check if what you actually have available before you use it (`useLocalInferenceTypePrint ` and `isolatedDeclarations` both act as discriminants) 

https://github.com/bloomberg/TypeScript/blob/f9aa1b6f93551019e8dea9d17c445818aac0452f/src/compiler/transformers/declarations.ts#L320